hazelcast/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -186,10 +192,31 @@ default boolean closeIsCooperative() {
      *              Note that it might not be the actual error that caused the job
      *              to fail - it can be several other exceptions. We only guarantee
      *              that it's non-null if the job didn't complete successfully.
+     * @see #isReusable()
      */
     default void close(@Nullable Throwable error) throws Exception {
     }
 
+    /**
+     * Returns {@code true} if this instance can be used in different vertices
+     * or in different job executions. In that case, {@link #init}, {@link #get}
+     * and {@link #close} methods must be idempotent and thread-safe.
+     * <p>
+     * When a job is submitted from a client, the job definition is serialized,
+     * so the job coordinator will receive a different copy of processor
+     * meta-suppliers even if they are used multiple times within the same DAG,
+     * or across different DAGs, or submitted through different jobs. While this
+     * serialization mechanism ensures that processor meta-suppliers do not have
```

Comment:
"serialization mechanism ensures that processor meta-suppliers do not have any internal state" - this is not precise